iroh-resolver/src/resolver.rs

```diff
@@ -480,6 +516,41 @@ impl<T: ContentLoader + Unpin + 'static> AsyncRead for OutPrettyReader<T> {
     }
 }
 
+impl<T: ContentLoader + Unpin + 'static> AsyncSeek for OutPrettyReader<T> {
+    fn start_seek(mut self: Pin<&mut Self>, position: std::io::SeekFrom) -> std::io::Result<()> {
+        match &mut *self {
+            OutPrettyReader::DagPb(bytes_reader)
+            | OutPrettyReader::DagCbor(bytes_reader)
+            | OutPrettyReader::DagJson(bytes_reader)
+            | OutPrettyReader::Raw(bytes_reader) => {
+                let pos_current = bytes_reader.pos;
+                let data_len = bytes_reader.bytes.len();
+                match position {
+                    std::io::SeekFrom::Start(pos) => {
+                        let i = std::cmp::min(data_len, pos as usize);
+                        bytes_reader.pos = i;
+                    }
+                    std::io::SeekFrom::End(_pos) => {
```

Comment:
I believe `End` also supports negative seek, eg `-5` which should be easily supported in this case